src/quacc/calculators/espresso/utils.py

```diff
@@ -0,0 +1,154 @@
+import warnings
+
+from ase.io.espresso import Namelist
+
+
+# Stolen from ase.io.espresso and modified to add a parameter 'keys',
+#  Maybe there is a better way of doing this?
```

Comment:
I'm not an expert on QE, but it would be nice to not rely on copying things so that upstream changes can be reflected here. I understand if that's not possible, however.